tests/OneTimePasswordTest.php

```diff
@@ -17,3 +17,14 @@
     expect($dto->getValidUntil())->toBe($validUntil);
     expect($dto->getValidityTimespan())->toBe($validityTimespan);
 });
+
+test('get valid from calculated correctly', function () {
```

Comment:
this test is wrong.

OTPs are only valid between seconds `0` and `29` or `30` to `59`. Every half minute, a new token needs to be calculated. In your test, the getValidFrom() has to be `2022, 8, 22, 13, 14, 30`.

for that reason, it is absolutely not necessary to have a ValidFrom sitting there. Everyone knows, that this token can only be created in second `0` or `30`. I would recommend to delete this part.